MSAL/src/native_auth/network/responses/validator/sign_in/MSALNativeAuthSignInResponseValidator.swift

```diff
@@ -94,8 +94,8 @@ final class MSALNativeAuthSignInResponseValidator: MSALNativeAuthSignInResponseV
             MSALLogger.log(
                 level: .error,
                 context: context,
-                format: "SignIn Challenge: Received unexpected challenge type: \(response.challengeType)")
-            return .error(.invalidServerResponse)
+                format: "signin/challenge: Received unexpected challenge type: \(response.challengeType)")
+            return .error(.unexpectedError(message: MSALNativeAuthErrorMessage.unexpectedErrorBody))
```

Comment:
This is not an `unexpectedErrorBody` so the message is not correct in this case

Reply:
See my other comment about potentially changing the `unexpectedErrorBody ` string. Or maybe a different string should be added for these cases where the challenge type is unexpected?

Reply:
Good catch 👀, done.